### PR TITLE
Changes for rpm and deb packaging

### PR DESCRIPTION
**Pull request by lynetteneo**
Additional changes to the spec file.
Added debian files for debian packaging for Ubuntu.
Includes previous changes to Makefile. For the previous mention of installing of service file in src/Makefile, there doesn't seem to be any issue when installing from the Makefile for Centos 7, Redhat 7, OpenSUSE Leap 15.2, or Ubuntu 18.04. These are the ones I've tried installing on and didn't face any issue running the Makefile to install the services. So I'm keeping the installing of the service file in src/Makefile instead of moving them to the spec file.

Signed-off-by: lynetteneo <lynette.sy.neo@seagate.com>

### CODE REVIEW

File: spec/seagate_ilm.spec [hunk 1]
```diff
@@ -0,0 +1,120 @@
+Summary: A lock manager based on Seagate In-Drive-Mutex (IDM)
+Name: seagate_ilm
+Epoch: 1
+Version: 1
+Release: 1%{?dist}
```
Comment (Leo-Yan) [inline spec/seagate_ilm.spec:5]: You could see that the propeller repo has a file VERSION, this file can be used to read out the version number.  And we can study sanlock to dynamically read out the release version by using the command:

~~~
ifeq ($(shell git describe --exact-match 2>/dev/null),)
# sanlock-3.7.0-5-g11fb098 -> 5.g11fb098
release := $(shell git describe --tags | awk -F- '{print $$(NF-1) "." $$(NF)}')
else
release := 0
endif
~~~

And you could see that sanlock uses a spec.in file, and Makefile can dynamically generate the sanlock.spec file based on sanlock.spec.in.  The benefit for this approach is we can only maintain the VERSION file and add new tag with "git tag" command, then the spec file can dynamically update its version number and release number based on the latest VERSION file ang git tag.

File: src/Makefile [hunk 2]
```diff
@@ -1,22 +1,16 @@
-# SPDX-License-Identifier: LGPL-2.1-only
-# Copyright (C) 2021 Seagate Technology LLC and/or its Affiliates.
-
+# Copyright 2020-2021 Seagate
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions
+# of the GNU General Public License v2 or (at your option) any later version.
+#
```
Comment (Leo-Yan) [inline src/Makefile]: This change is unexpected, should remove it.

File: src/Makefile [hunk 3]
```diff
@@ -1,22 +1,16 @@
...
 
-VER := $(shell cat ../VERSION)
-
-GIT_INFO := $(shell git rev-parse --is-inside-work-tree >/dev/null; echo $$?)
-ifeq ($(GIT_INFO),0)
-VER := "$(shell git describe --abbrev=12 --dirty --always --tags)"
-endif
-
-CFLAGS += -DVERSION=\"$(VER)\"
-
```
Comment (Leo-Yan) [inline src/Makefile]: This change is unexpected, should remove it.

File: src/Makefile [hunk 4]
```diff
@@ -33,9 +27,7 @@ SOURCE = main.c \
 	 raid_lock.c \
 	 inject_fault.c \
 	 failure.c \
-	 drive.c \
-	 scsiutils.c \
-	 uuid.c
+	 drive.c
```
Comment (Leo-Yan) [inline src/Makefile]: This change is unexpected, should remove it.

File: src/Makefile [hunk 5]
```diff
@@ -78,12 +70,15 @@ HEADIR=/usr/include
...
 	$(INSTALL) -c -m 755 $(TARGET) $(DESTDIR)/$(BINDIR)
 	$(INSTALL) -c -m 755 $(LIBSO_TARGET) $(DESTDIR)/$(LIBDIR)
 	$(INSTALL) -c -m 644 $(LIBPC_TARGET) $(DESTDIR)/$(LIBDIR)/pkgconfig
 	cp -a $(LIB_TARGET).so $(DESTDIR)/$(LIBDIR)
 	cp -a $(LIB_TARGET).so.$(SOMAJOR) $(DESTDIR)/$(LIBDIR)
 	$(INSTALL) -c -m 644 $(HEADER_TARGET) $(DESTDIR)/$(HEADIR)
-	$(INSTALL) -c -m 644 seagate_ilm.service /usr/lib/systemd/system/
-	$(INSTALL) -c -m 644 logrotate.ilm /etc/logrotate.d/seagate_ilm
+	$(INSTALL) -c -m 644 seagate_ilm.service $(DESTDIR)/usr/lib/systemd/system/
+	$(INSTALL) -c -m 644 logrotate.ilm $(DESTDIR)/etc/logrotate.d/seagate_ilm
```
Comment (Leo-Yan) [inline src/Makefile]: The change looks good to me.

File: src/Makefile [hunk 6]
```diff
@@ -1,16 +1,22 @@
-# Copyright 2020-2021 Seagate
-#
-# This copyrighted material is made available to anyone wishing to use,
-# modify, copy, or redistribute it subject to the terms and conditions
-# of the GNU General Public License v2 or (at your option) any later version.
-#
+# SPDX-License-Identifier: LGPL-2.1-only
+# Copyright (C) 2021 Seagate Technology LLC and/or its Affiliates.
+
```
Comment (Leo-Yan) [inline src/Makefile]: This patch should be dropped because this patch is to recovery the unexpected changes in the previous patch.

**Comment by Leo-Yan** [comment]
Hi @lynetteneo, the changes look good to me, except I give a suggestion for how to dynamically generate version number and release tag based on the VERSION file and git tag.

@daniel-thompson could you give a review and shall we ask any Linaro colleague with releasing expertise to give a review, especially for the Debian releasing files?

File: src/Makefile (hunk 6 above)
Comment (lynetteneo) [reply src/Makefile]: Still not sure how to use GitHub. I've created another commit after this commit to change back the Makefile. Or should I be creating another pull request entirely?

File: spec/seagate_ilm.spec (hunk 1 above)
Comment (lynetteneo) [reply spec/seagate_ilm.spec:5]: I'll take a look at this.

File: src/Makefile (hunk 6 above)
Comment (Leo-Yan) [reply src/Makefile]: You don't need to create another new branch, using the command "git push -f" to upload the updated branch onto your github repositorty, the pull request can reflect the updated changes.

**Review by daniel-thompson — APPROVED** [review]
There are some minor things that could be improved but nothing that I think argues against merging.

Firstly the first commit would arguable be better as three. 1. Fix the DESTDIR bug in the Makefile, 2. Update rpm spec file, 3. Add deb packaging.

Secondly there is no scripting in the Debian package to ensure the service gets enabled. Normally a debian package has debconf integration to ensure the user is asked whether to enable the systemd service when the package is installed (user interaction during package install/upgrade is *the* big difference between RPM and deb packaging philosophy... RPM does sane defaults and no user questions, deb does sane defaults but asks user if they argee).

To be clear the scripting should be added but I think that could go on your TODO list and need not stop merging the change.

File: Makefile [hunk 7]
```diff
@@ -23,3 +31,8 @@ cscope:
 	@echo "  CSCOPE"
 	${Q}find ${CURDIR} -name "*.[ch]" > cscope.files
 	${Q}cscope -b -q -k
+
+spec:
+	sed -e 's/@VERSION@/$(version)/g' \
+		-e 's/@RELEASE@/$(release)/g' \
+		seagate_ilm.spec.in > seagate_ilm.spec
```
Comment (daniel-thompson) [inline Makefile:38]: Shouldn't there be equivalent changes for Debian packaging?

**Pull request merged**